tool/clean/clean_security_group/clean_security_group_test.go

```diff
@@ -0,0 +1,298 @@
+	originalDryRun := cfg.dryRun
+	defer func() { cfg.dryRun = originalDryRun }()
+
+	tests := []struct {
+		name                 string
+		securityGroup        types.SecurityGroup
+		dryRun               bool
+		hasNetworkInterfaces bool
+		expectDelete         bool
+		expectRevokeRules    bool
+	}{
+		{
+			name: "Default security group - should not delete",
+			securityGroup: types.SecurityGroup{
+				GroupId:   aws.String("sg-12345"),
+				GroupName: aws.String("default"),
+			},
+			dryRun:               false,
+			hasNetworkInterfaces: false,
+			expectDelete:         false,
+			expectRevokeRules:    false,
+		},
+		{
+			name: "Security group with network interfaces - should not delete",
+			securityGroup: types.SecurityGroup{
+				GroupId:   aws.String("sg-67890"),
+				GroupName: aws.String("test-group"),
+			},
+			dryRun:               false,
+			hasNetworkInterfaces: true,
+			expectDelete:         false,
+			expectRevokeRules:    false,
+		},
+		{
+			name: "Unused security group with rules - should delete and revoke rules",
+			securityGroup: types.SecurityGroup{
+				GroupId:   aws.String("sg-abcde"),
+				GroupName: aws.String("test-group"),
+				IpPermissions: []types.IpPermission{
+					{
+						IpProtocol: aws.String("tcp"),
+						FromPort:   aws.Int32(22),
+						ToPort:     aws.Int32(22),
+					},
+				},
+				IpPermissionsEgress: []types.IpPermission{
+					{
+						IpProtocol: aws.String("-1"),
+					},
+				},
+			},
+			dryRun:               false,
+			hasNetworkInterfaces: false,
+			expectDelete:         true,
+			expectRevokeRules:    true,
+		},
+		{
+			name: "Dry run - should not actually delete",
+			securityGroup: types.SecurityGroup{
+				GroupId:   aws.String("sg-abcde"),
+				GroupName: aws.String("test-group"),
+			},
+			dryRun:               true,
+			hasNetworkInterfaces: false,
+			expectDelete:         false,
+			expectRevokeRules:    false,
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			cfg.dryRun = tt.dryRun
+
+			// Create mock client
+			mockClient := &mockEC2Client{
+				describeNetworkInterfacesOutput: &ec2.DescribeNetworkInterfacesOutput{
+					NetworkInterfaces: make([]types.NetworkInterface, 0),
+				},
+				describeSecurityGroupsOutput: &ec2.DescribeSecurityGroupsOutput{
+					SecurityGroups: []types.SecurityGroup{tt.securityGroup},
+				},
+			}
+
+			if tt.hasNetworkInterfaces {
+				mockClient.describeNetworkInterfacesOutput.NetworkInterfaces = append(
+					mockClient.describeNetworkInterfacesOutput.NetworkInterfaces,
+					types.NetworkInterface{
+						NetworkInterfaceId: aws.String("eni-12345"),
+					},
+				)
+			}
+
+			// Create worker
+			w := worker{
+				id:                       1,
+				wg:                       &sync.WaitGroup{},
+				deletedSecurityGroupChan: make(chan string, 1),
+			}
+
+			// Call the function
+			err := w.handleSecurityGroup(context.Background(), mockClient, tt.securityGroup)
```

Comment:
Maybe we can output the err here to see wy the test failed
```suggestion
            	err := w.handleSecurityGroup(context.Background(), mockClient, tt.securityGroup)
		if err != nil {
		                t.Fatalf("Unexpected error: %v", err)
		            }```

Reply:
Doesn't no error print it out if there is an error?